packages/angular_devkit/build_angular/src/webpack/plugins/karma/karma.ts

```diff
@@ -86,32 +86,11 @@ const init: any = (config: any, emitter: any) => {
   config.reporters.unshift('@angular-devkit/build-angular--event-reporter');
 
   // When using code-coverage, auto-add karma-coverage.
-  if (options.codeCoverage) {
-    config.plugins = config.plugins || [];
-    config.reporters = config.reporters || [];
-    const { plugins, reporters } = config;
-    const hasCoveragePlugin = plugins.some(isPlugin('karma-coverage', 'reporter:coverage'));
-    const hasIstanbulPlugin = plugins.some(
-      isPlugin('karma-coverage-istanbul-reporter', 'reporter:coverage-istanbul'),
-    );
-    const hasCoverageReporter = reporters.includes('coverage');
-    const hasIstanbulReporter = reporters.includes('coverage-istanbul');
-    if (hasCoveragePlugin && !hasCoverageReporter) {
-      reporters.push('coverage');
-    } else if (hasIstanbulPlugin && !hasIstanbulReporter) {
-      // coverage-istanbul is deprecated in favor of karma-coverage
-      reporters.push('coverage-istanbul');
-    } else if (!hasCoveragePlugin && !hasIstanbulPlugin) {
-      throw new Error('karma-coverage must be installed in order to run code coverage.');
```

Comment:
Is this plugin check still useful?

Reply:
Need to check if Karma throw a meaningful error when not installed.

Reply:
There is an error message, but it is pretty [generic](https://github.com/karma-runner/karma/blob/2b71a3c6a01757e0b5c1bf8d241b588656b62127/lib/plugin.js#L85). Probably better to have a dedicated error message in the CLI with the specific migration instructions or an update schematic as the switch requires both dependency changes and [configuration changes](https://github.com/angular/angular-cli/commit/8995e49f0bc2de3cba2f2fea87afe898f7bedf71#diff-f23d77c1515d2212caa684c6edee8c197a846c385405851ea9be3bd90f3179f9) (i.e. just installing karma-coverage is not enough).

Reply:
I think `Can not load reporter "coverage", it is not registered! Perhaps you are missing some plugin?` is good enough. 

Regarding the migration. As far as I reckon back in version 11 we decided that we'll it up to the users to migrate as it was non trivial.